File: app/src/main/java/tomaszkopacz/meetbam/di/components/WebServiceComponent.java
package tomaszkopacz.meetbam.di.components;

import javax.inject.Singleton;

import dagger.Component;
import tomaszkopacz.meetbam.activities.LoginActivity;
import tomaszkopacz.meetbam.activities.MainActivity;
import tomaszkopacz.meetbam.di.providers.WebServiceModule;

/**
 * Created by tomas on 03.03.2018.
 */

@Singleton
@Component(modules = WebServiceModule.class)
public interface WebServiceComponent {

    void inject(MainActivity mMainActivity);
    void inject(LoginActivity mLoginActivity);
}

File: app/src/main/java/tomaszkopacz/meetbam/di/WebServiceComponent.kt
package tomaszkopacz.meetbam.di

import dagger.Component
import tomaszkopacz.meetbam.presenter.*
import javax.inject.Singleton

@Singleton
@Component(modules = [(WebServiceModule::class)])
interface WebServiceComponent {

    fun inject(mainActivityPresenter: MainPostsFragmentPresenter)
    fun inject(mLoginActivityPresenter: LoginActivityPresenter)

    fun inject(mAccountPhotosFragmentPresenter: AccountPhotosFragmentPresenter)
    fun inject(mAccountFriendsFragmentPresenter: AccountFriendsFragmentPresenter)

    fun inject(mStatsGlobalFragmentPresenter: StatsGlobalFragmentPresenter)
    fun inject(mStatsLocalFragmentPresenter: StatsLocalFragmentPresenter)
}
